### PR TITLE
chore: replace goKitLogger in k8sutil and event handler

### PR DESCRIPTION
**Pull request by perebaj**
## Description

Replace goKitLogger in k8sutil package and EventHandler struct 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: pkg/k8sutil/secrets_test.go [hunk 1]
```diff
@@ -39,7 +41,10 @@ func TestLoadSecretRef(t *testing.T) {
 	}
 
 	sClient := fake.NewSimpleClientset(secret).CoreV1().Secrets("ns")
-	logger := log.NewNopLogger()
+	logger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{
+		// slog level math.MaxInt means no logging
+		Level: slog.Level(math.MaxInt),
+	}))
```
Comment (perebaj) [inline pkg/k8sutil/secrets_test.go:47]: The previous logger isn't logging nothing, for this reason, I'm using `Level: sloglLevel(math.MaxInt)`

Comment (ArthurSens) [reply pkg/k8sutil/secrets_test.go:47]: Just did a little research and looks like slog doesn't have a builtin No-op logger. It's in the works tho (See https://github.com/golang/go/issues/62005) :)

Could you add a comment here saying that we want to use slog's builtin No-op logger once it's available? You can add the Go's issue link to make it clearer

Comment (perebaj) [reply pkg/k8sutil/secrets_test.go]: good catch! tnks.

[079a260](https://github.com/prometheus-operator/prometheus-operator/pull/6832/commits/079a2602d650a5c8e2ef7f8c6815c5d1d7fb7085)

We are using this level across other parts of the code, in the next PR I will address them putting a similar comment

**Review by ArthurSens — APPROVED** [review]
Looks good, thanks!

**Pull request merged**